dlt-database/entity/0003-refactor_transaction_recipe/Account.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  ManyToOne,
+  JoinColumn,
+  OneToMany,
+  BaseEntity,
+} from 'typeorm'
+import { User } from '../User'
+import { Transaction } from '../Transaction'
+import { DecimalTransformer } from '../../src/typeorm/DecimalTransformer'
+import { Decimal } from 'decimal.js-light'
+import { AccountCommunity } from '../AccountCommunity'
+
+@Entity('accounts')
+export class Account extends BaseEntity {
+  @PrimaryGeneratedColumn('increment', { unsigned: true })
+  id: number
+
+  @ManyToOne(() => User, (user) => user.accounts, { cascade: ['insert'], eager: true }) // Assuming you have a User entity with 'accounts' relation
+  @JoinColumn({ name: 'user_id' })
+  user?: User
+
+  // if user id is null, account belongs to community gmw or auf
+  @Column({ name: 'user_id', type: 'int', unsigned: true, nullable: true })
+  userId?: number
+
+  @Column({ name: 'derivation_index', type: 'int', unsigned: true, nullable: true })
+  derivationIndex?: number
+
+  @Column({ name: 'derive2_pubkey', type: 'binary', length: 32, unique: true })
+  derive2Pubkey: Buffer
+
+  @Column({ type: 'tinyint', unsigned: true })
+  type: number
+
+  @Column({
+    name: 'created_at',
+    type: 'datetime',
+    precision: 3,
+    default: () => 'CURRENT_TIMESTAMP(3)',
+  })
+  createdAt: Date
+
+  @Column({ name: 'confirmed_at', type: 'datetime', nullable: true })
+  confirmedAt?: Date
+
+  @Column({
+    type: 'decimal',
+    precision: 40,
+    scale: 20,
+    default: 0,
+    transformer: DecimalTransformer,
+  })
+  balance: Decimal
+
+  @Column({
+    name: 'balance_date',
+    type: 'datetime',
+    default: () => 'CURRENT_TIMESTAMP()',
+  })
+  balanceDate: Date
+
+  @Column({
+    name: 'balance_created_at',
+    type: 'decimal',
+    precision: 40,
+    scale: 20,
+    default: 0,
+    transformer: DecimalTransformer,
+  })
+  balanceCreatedAt: Decimal
+
+  @Column({
+    name: 'balance_created_at_date',
+    type: 'datetime',
+    precision: 3,
+    default: () => 'CURRENT_TIMESTAMP(3)',
+  })
+  balanceCreatedAtDate: Date
```

Comment:
these four columns are confusing or not clear for me.

What are the intentional difference between:
- balanceDate and balanceCreatedAtDate
- balance and balanceCreatedAt

The naming patterns confusing me:
- ending with `Date` suggerates a `datetime` column
- but ending with `CreatedAt` suggerates in all other entities also a `datetime`column and not a `Decimal`

Should these column-pairs represent the balance value and time before and after iota-validation?

Reply:
You are right that wasn't clear, I have updated it to:
```ts
@Column({
    name: 'balance_confirmed_at',
    type: 'decimal',
    precision: 40,
    scale: 20,
    default: 0,
    transformer: DecimalTransformer,
  })
  balanceConfirmedAt: Decimal

  @Column({
    name: 'balance_confirmed_at_date',
    type: 'datetime',
    default: () => 'CURRENT_TIMESTAMP()',
  })
  balanceConfirmedAtDate: Date

  @Column({
    name: 'balance_created_at',
    type: 'decimal',
    precision: 40,
    scale: 20,
    default: 0,
    transformer: DecimalTransformer,
  })
  balanceCreatedAt: Decimal

  @Column({
    name: 'balance_created_at_date',
    type: 'datetime',
    precision: 3,
    default: () => 'CURRENT_TIMESTAMP(3)',
  })
  balanceCreatedAtDate: Date
  ```
So it is the balance calculated at creation date, like the backend do it
and balance calculated at confirmation date (date of iota milestone, seconds precision) like GradidoNode do it. 

I had no better idea for the naming pattern for this. Maybe you can think of something even better.

Reply:
The entity already has a pair of create_at and confirmed_at dates for his own creation and confirmation.